var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -188,8 +188,9 @@ def flag_handler(self, name, flags):
         cmake_flags = []
 
         if name == "cflags":
-            cc_name = os.path.basename(self.compiler.cc)
-            if "clang" in cc_name or "gcc" in cc_name:
+            if self.spec.satisfies('%gcc') \
+                    or self.spec.satisfies('%clang') \
+                    and not self.spec.satisfies('%nvhpc'):
```

Comment:
I'm not sure I understand this logic. Shouldn't `self.spec.satisfies('%gcc') or self.spec.satisfies('%clang')` suffice? It can't be `%gcc` and `%nvhpc` at the same time, right?

Reply:
You're correct, I wrote it that way for emphasis, because the change is needed to build for nvhpc, so the phrasing is more literary than logical. Logically, not %nvhpc is implicit. Feel free to change if you think the phrasing should be minimal.

Reply:
I think it would be easier to understand if it was minimal.